File: JavaScript/factorial_recursive.js
function main(){
    let number = 6;
    console.log(factorial_recursive(number));
}

function factorial_recursive(number){
    if (number <= 1){
        return 1;   // Base case
    } else {
        return number * factorial_recursive(number - 1);
    }
}

main();

File: Python/factorial_recursive.py
def factorial_recursive(n):
	"""
	Description:
		Calculate and return the factorial of 'n' (n!);
		using a recusive approach.
		Time complexity: O(n)
	
	Arguments:
		n: Integer >= 0
	
	Returns:
		int: Factorial of 'n' (n!)
	"""
	assert isinstance(n, int) and n >= 0, "n must be positive int"
	
	# Find n! using recursion.
	if n <= 1:
		return 1
	else:
		return n * factorial_recursive(n - 1)